### ISSUE TITLE
Issue while backpropagating through sparse_tensor_dense_multiply

### ISSUE THREAD

**Issue by saisrivatsan** [tensorflow/tensorflow#11036]
I have a simple network defined as follows:

    h1 = tf.sparse_tensor_dense_matmul(x, W1)
    h2 = tf.matmul(h1, W2)
    y = tf.matmul(h2, W3)
    loss = tf.nn.l2_loss(y - y_)
    train = tf.train.AdamOptimizer(learning_rate=lr).minimize(loss)

where x is a sparseTensor, rest are dense. 
Dimensions (shapes) of W1 = [1000,200], W2 = [200,400] and W3 = [ 400, 500]. 

When I run the following:

    sess.run([train], feed_dict={x:X, y_:Y})

where X is sparseTensor of shape [N, 1000] and Y is a tensor of shape [N, 500]

I get an error saying: OOM when allocating tensor with shape[3684773,200].
This is happening while the the the gradient for W is being computed. 3684773 also happens to be the number of non-zero elements in X.

Note:

 1. When I compute gradients using tf.gradients, they work completely
    fine.
 2. When I run the same network using dense X and dense multiply( tf.matmul ), it works completely fine.

**Comment by ali01** [comment]
Please provide details about what platform you are using  (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?  Make sure you also include the exact command if possible to produce  the output included in your test case. If you are unclear what to include  see the issue template displayed in  [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

 We ask for this in the issue submission template, because    it is really difficult to help without that information. Thanks!

**Comment by tensorflowbutler** [comment]
This issue is automatically closed due to lack of activity. Please re-open if this is still an issue for you. Thanks!
